Fix warnings

ret2libc opened the pull request.

radare commented on libr/asm/arch/arm/armass.c:
```diff
@@ -665,8 +665,12 @@ static st32 getlistmask(char *input) {
-		temp[i] = 0;
+		if (i > 0) {
+			strncpy (temp, input, i);
+			temp[i] = 0;
```
be consistent, what about making this temp[i] = 0; once, by using R_MAX(i, 0)

radare commented on libr/core/visual.c:
```diff
@@ -769,25 +769,28 @@ static void findPrevWord(RCore *core) {
 	const ut8 *p;
 	int len, d = core->print->cur;
-	char str[128], buf[258];
+	char str[128], buf[BUF_SZ];
```
dont use defines to assume the size of a buffer, this may lead to inconsistencies if not refactored properly. it is recommended to u se sizeof()

radare commented on libr/core/visual.c:
```diff
@@ -769,25 +769,28 @@ static void findPrevWord(RCore *core) {
-			}
-		}
+		strncpy (buf, str + 1, sizeof (buf) - 1);
+		buf[BUF_SZ - 1] = '\0';
```
use siezof

radare commented on libr/anal/esil2reil.c:
```diff
@@ -59,7 +59,8 @@ RAnalReilArg *reil_pop_arg(RAnalEsil *esil) {
 		if (buf[i] == ':') {
 			tmp_buf[j] = '\0';
-			strncpy (op->name, tmp_buf, sizeof (op->name) - 1);
+			strncpy (op->name, tmp_buf, sizeof (op->name));
```
use r_str_ncpy

radare commented on libr/anal/esil2reil.c:
```diff
@@ -76,8 +77,9 @@ RAnalReilArg *reil_pop_arg(RAnalEsil *esil) {
 	if (!flag) {
-		strncpy(op->name, tmp_buf, sizeof(op->name) - 1);
-		op->type = reil_get_arg_type(esil, op->name);
+		strncpy(op->name, tmp_buf, sizeof(op->name));
```
use r_str_ncpy

radare commented on libr/asm/arch/arm/armass.c:
```diff
@@ -665,8 +665,12 @@ static st32 getlistmask(char *input) {
 		}
-		strncpy (temp, input, i);
-		temp[i] = 0;
+		if (i > 0) {
```
using r_str_ncpy  should be  enough

radare commented on libr/core/visual.c:
```diff
@@ -769,25 +769,28 @@ static void findPrevWord(RCore *core) {
+			len--;
+		}
 	} else if (len < 1) {
 		strncpy (buf, str, sizeof (buf) - 1);
```
use r_str_ncpy

radare commented on libr/fs/shell.c:
```diff
@@ -119,6 +119,7 @@ R_API int r_fs_shell_prompt(RFSShell* shell, RFS* fs, const char* root) {
 		} else if (!memcmp (buf, "cd ", 3)) {
 			char opath[PROMPT_PATH_BUFSIZE];
 			strncpy (opath, path, sizeof (opath) - 1);
+			opath[sizeof (opath) - 1] = '\0';
```
use r_str_ncpy

ret2libc commented:
Ok I'm using r_str_ncpy now. I've also changed a bit the signature to use `size_t` instead of `int` for the `n` parameter. Moreover, I added a check for n==0, to prevent weird behaviours.
